### PR TITLE
[FIX] mail: Fixed sending  of 'email' when email type <> 'email' and/o...

### LINKED ISSUE #1566: [mail] Not work forwarding email
Impacted versions:
* 7.0 and above

Steps to reproduce:
* To set an outgoing mail server causing some error in "Test connection", in this case configured wrong pasword
outgoing mail servers openerp

* To create new partner "test"
customers openerp
* To create new sale order with customer "test" and send by email
sales orders openerp
*  Go to emails and we can observed that email status is delivery failed
emails openerp
* Set correctly the outgoing email server
outgoing mail servers openerp
* Go to emails and try to forward email again
emails openerp

Current behavior:
* The email is not sent

Expected behavior:
* The email should be sent

### PR DESCRIPTION
**Pull request by rgo-odoo**
...r no value for 'recipient_ids' param for send(...).(maintenance: 611701)

Fixes #1566
Replace #2307

### CODE REVIEW

**Comment by mart-e** [comment]
Hello,

Your patch will work in the reported scenario but not in the following:
1. send an email to 10 people (with one having an invalid email)
2. the email is sent to 9 people, log a warning and is put in delivery failed
3. when retried manually (after correction of the email), it will be sent to 10 people, 9 having already received the email

For this reason we can not merge it in 7.0. In 8.0 the list of emails is stored on the email but not in 7.0 where we have to recompute it via `get_partners_to_notify`.

Long story short: can not properly fix in 7.0 without backporting new fields.
